Issue in imf-reader dependency

Hi all,

(apologies in advance for the long error trace).

while using pydeflate (version 2.3.0), I encountered the following issue.

```
.venv/lib/python3.13/site-packages/pydeflate/deflate/deflators.py:74: in wrapper
    source = deflator_source_cls(update=update_deflators)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.13/site-packages/pydeflate/core/source.py:125: in __init__
    super().__init__(name="IMF", reader=read_weo, update=update)
<string>:7: in __init__
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = IMF(name='IMF', reader=<function read_weo at 0x117190c20>, update=False, data=Empty DataFrame
Columns: []
Index: [])

    def __post_init__(self):
        """Load and validate data after initialization."""
        try:
            self.data = self.reader(self.update)
        except Exception as e:
>           raise DataSourceError(
                f"Failed to load data: {e}",
                source=self.name,
            ) from e
E           pydeflate.exceptions.DataSourceError: [IMF] Failed to load data: Could not fetch data for version: April 2025. File is not a zip file
```

It seems to me that the errore is related to an error in the `imf-reader` package. 

This is because I also get the error

```
2025-11-18 13:15:45,634 - pydeflate - INFO:
 Downloading the latest IMF WEO dataset...
INFO: No data found for expected latest version: October 2025. Rolling back version...
INFO: No data found for expected latest version: October 2025. Rolling back version...
------------------------------ Captured log call -------------------------------
INFO     imf_reader.config:reader.py:163 No data found for expected latest version: October 2025. Rolling back version...

    @staticmethod
    def get_sdmx_url(soup: BeautifulSoup) -> str:
        """Get the url to download the WEO data in SDMX format.
    
        Args:
            soup: BeautifulSoup object of the IMF WEO website.
    
        Returns:
            The url to download the SDMX data.
        """
    
        try:
>           href = soup.find("a", string="SDMX Data").get("href")
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E           AttributeError: 'NoneType' object has no attribute 'get'
.venv/lib/python3.13/site-packages/imf_reader/weo/scraper.py:49: AttributeError
During handling of the above exception, another exception occurred:
version = ('October', 2025)
    def fetch_data(version: Optional[Version] = None) -> pd.DataFrame:
        """Fetch WEO data
    
        By default, this function fetched data for the latest WEO publication. If a specific publication version
        is required, the version can be passed as a tuple of month and year. WEO data is released in April and October
        each year. For the version month, the month must be either "April" or "October"
        This function caches the data for faster access and to prevent multiple requests to the IMF website. To clear the
        cache, use the `clear_cache` function.
    
        e.g.
        >>> fetch_data() # fetches the latest data
        >>> fetch_data(("April", 2024)) # fetches the data for April 2024
During handling of the above exception, another exception occurred:
version = None
    def fetch_data(version: Optional[Version] = None) -> pd.DataFrame:
        """Fetch WEO data
    
        By default, this function fetched data for the latest WEO publication. If a specific publication version
        is required, the version can be passed as a tuple of month and year. WEO data is released in April and October
        each year. For the version month, the month must be either "April" or "October"
        This function caches the data for faster access and to prevent multiple requests to the IMF website. To clear the
        cache, use the `clear_cache` function.
    
        e.g.
        >>> fetch_data() # fetches the latest data
        >>> fetch_data(("April", 2024)) # fetches the data for April 2024
    
        Args:
            version: The version of the WEO data to fetch as a tuple eg `("April", 2023).
                     By default, the latest version is fetched.
    
        Returns:
            A pandas DataFrame containing the WEO data
        """
    
        # if version is passed, validate it and fetch the data
        if version is not None:
            try:
                version = validate_version(version)
                df = _fetch(version)
                fetch_data.last_version_fetched = (
                    version  # store the version fetched as function attribute
                )
                return df
            except Exception as e:
                raise NoDataError(
                    f"Could not fetch data for version: {version[0]} {version[1]}. {str(e)}"
                )
    
        # if no version is passed, generate the latest version and fetch the data
        latest_version = gen_latest_version()
        try:
>           return fetch_data(latest_version)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^
```

I am happy to provide further details if needed.

Thanks!

FYI @euronion

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,10 @@ All notable changes to this project are documented in this file. The format foll
 
 ## [Unreleased]
 
+## [2.3.1] - 2025-12-05
+### Changed
+- Increased minimum `imf-reader` version to 1.4.1 to support the new IMF SDMX API for fetching latest data.
+
 ## [2.3.0] - 2025-10-05
 ### Added
 - **Get Deflators Functions**: New functions to retrieve deflator data directly as DataFrames without requiring user data:

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [project]
 name = "pydeflate"
-version = "2.3.0.post0"
+version = "2.3.1"
 description = "Package to convert current prices figures to constant prices and vice versa"
 readme = "README.md"
 requires-python = ">=3.11"
@@ -11,7 +11,7 @@ license-files = ["LICENSE*"]
 
 dependencies = [
     "hdx-python-country>=3.9.8",
-    "imf-reader>=1.3.0",
+    "imf-reader>=1.4.1",
     "oda-reader>=1.3.0",
     "pandas>=2.0",
     "pandera>=0.20.0",

--- a/src/pydeflate/deflate/get_deflators.py
+++ b/src/pydeflate/deflate/get_deflators.py
@@ -72,7 +72,9 @@ def _get_deflator(deflator_source_cls, price_kind):
             data = deflator.pydeflate_data.copy()
 
             # Determine the entity column based on use_source_codes
-            entity_col = "pydeflate_entity_code" if use_source_codes else "pydeflate_iso3"
+            entity_col = (
+                "pydeflate_entity_code" if use_source_codes else "pydeflate_iso3"
+            )
 
             # Filter by countries if specified
             if countries is not None:

--- a/src/pydeflate/exchange/get_rates.py
+++ b/src/pydeflate/exchange/get_rates.py
@@ -66,7 +66,9 @@ def _get_exchange_rates(exchange_source_cls, **fixed_params):
             data = exchange.pydeflate_data.copy()
 
             # Determine the entity column based on use_source_codes
-            entity_col = "pydeflate_entity_code" if use_source_codes else "pydeflate_iso3"
+            entity_col = (
+                "pydeflate_entity_code" if use_source_codes else "pydeflate_iso3"
+            )
 
             # Filter by countries if specified
             if countries is not None:

--- a/tests/integration/test_get_deflators.py
+++ b/tests/integration/test_get_deflators.py
@@ -104,9 +104,7 @@ def test_get_deflators_consistency_with_deflate(sample_source_frames):
     )
 
     # Create test data
-    test_data = pd.DataFrame(
-        {"iso_code": ["USA"], "year": [2021], "value": [100.0]}
-    )
+    test_data = pd.DataFrame({"iso_code": ["USA"], "year": [2021], "value": [100.0]})
 
     # Deflate using the regular function
     deflated = imf_gdp_deflate(data=test_data, base_year=2022, value_column="value")

--- a/tests/integration/test_get_rates.py
+++ b/tests/integration/test_get_rates.py
@@ -75,9 +75,9 @@ def test_get_exchange_rates_consistency_with_exchange(sample_source_frames):
     )
 
     # Get the exchange rate value
-    rate_value = rates[
-        (rates["iso_code"] == "FRA") & (rates["year"] == 2021)
-    ]["exchange_rate"].iloc[0]
+    rate_value = rates[(rates["iso_code"] == "FRA") & (rates["year"] == 2021)][
+        "exchange_rate"
+    ].iloc[0]
 
     # The exchanged value should equal original * rate
     expected = 100.0 * rate_value

